src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,200 @@
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Obtain the true first index
+    startIndex(): number {
+        return Math.max(this.firstIndex, this.seq.first)
+    }
+
+    // Obtain the true number of terms
+    numTerms(): number {
+        return Math.min(this.terms + this.startIndex(), this.seq.last)
+    }
+
+    // Find the largest value of the sequence asked for
+    largestValue(): bigint {
+        let largest_value = this.seq.getElement(this.startIndex())
+        for (let i = this.startIndex(); i < this.numTerms(); i++) {
+            const value = this.seq.getElement(i)
+            if (value > largest_value) {
+                largest_value = value
+            }
+        }
+        return largest_value
+    }
+
+    // Create an array with the number of factor of
+    // each element at the corresponding index of the array
+    factorArray(): number[] {
+        const factorArray = []
+        for (let i = this.startIndex(); i < this.numTerms(); i++) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [base, power] of factors) {
+                    if (base > 0) {
+                        counter += Number(power)
+                    } else if (base === 0n) {
+                        counter = 0
+                    }
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding bins
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            binFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            binFactorArray[Math.trunc(factorArray[i] / this.binSize)]++
+        }
+
+        return binFactorArray
+    }
+
+    // Create a number that represents how
+    // many pixels wide each bin should be
+    binWidth(): number {
+        const binFactorArray = this.binFactorArray()
+        let array_length = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] != 0) {
+                array_length = i
+            }
+        }
+        return (0.95 * this.sketch.width) / (array_length + 1)
+    }
+
+    // Create a number that represents how many pixels high
+    // each increase of one in the bin array should be
+    height(): number {
+        const binFactorArray = this.binFactorArray()
+        let greatest_value = binFactorArray[0]
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] > greatest_value) {
+                greatest_value = binFactorArray[i]
+            }
+        }
```

Comment:
And it looks like you will have repeated code for finding the largest value in an array (largest in factorArray to know the number of bins, largest in binFactorArray to know how many pixels per unit height) so you should make yourself a utility function for that, rather than having essentially the same code twice. DRY and all.